### ISSUE TITLE
Illegal instruction on Julia v1.2-rc1 (regression)

### ISSUE THREAD

**Issue by oschulz** [JuliaLang/julia#32386]
On Julia v1.2-rc1 and v1.3.0-DEV.433 (2019-06-20), we get an `Illegal instruction` error with [SolidStateDetectors](https://github.com/JuliaPhysics/SolidStateDetectors.jl):

```
(v1.2) pkg> add TypedTables#master SolidStateDetectors
(v1.2) pkg> test SolidStateDetectors
   Testing SolidStateDetectors
 Resolving package versions...
 [...]
Unreachable reached at 0x7f75351968fa

signal (4): Illegal instruction
in expression starting at [...]/packages/SolidStateDetectors/7T4zz/test/runtests.jl:8
construct_semiconductor at [...]/packages/SolidStateDetectors/7T4zz/src/SolidStateDetector/SolidStateDetector.jl:72
jl_apply_generic at /buildworker/worker/package_linux64/build/src/gf.c:2197
construct_objects at [...]/packages/SolidStateDetectors/7T4zz/src/SolidStateDetector/SolidStateDetector.jl:86
jl_apply_generic at /buildworker/worker/package_linux64/build/src/gf.c:2197
[...]
```

Note: Needs TypedTables#master until a new version of TypedTables is released (see FugroRoames/TypedTables.jl#52).

On Julia v1.1.1 all is fine.

Not sure if this is related to #32222 - it also happens with the current nightly though, which has #32222 fixed.

We've (@oschulz, @lmh91, @hervasa2) run into this on Ubuntu 16, Ubuntu 19, CentOS-7 and OS-X, so doesn't seem to be OS-related.

**Comment by JeffBezanson** [comment]
This is slightly similar to #32222 in that I can't immediately figure out what's going on...

**Comment by oschulz** [comment]
Sorry that we didn't provide an MWE, didn't manage to put one together so far that reproduces the error. Will be above be Ok for you to track it down?

**Comment by JeffBezanson** [comment]
Yes, the above is enough.

**Comment by oschulz** [comment]
Thanks, Jeff!

**Comment by JeffBezanson** [comment]
Seems to be caused by e3de220fa22f06d44fb7e2b3ba9f090696fbfd84. Looks like this is a simpler issue than #32222 after all.

**Comment by oschulz** [comment]
Ah,  e3de220 isn't part of v1.2-rc1, right? Will it go into rc2?

**Comment by JeffBezanson** [comment]
No I believe it is in 1.2-rc1.

**Comment by oschulz** [comment]
Ah, sorry, silly me - of course it is, otherwise we wouldn't have run into this issue, after all.

**Comment by oschulz** [comment]
So I guess fix #31993 needs a fix itself? :-)

Sorry I can't be of any help, here, I'm really not familiar enough with the internals of the Julia compiler ...

**Comment by oschulz** [comment]
Thanks a lot for the speedy fix, @JeffBezanson!
